web/src/integration-test/java/ca/hendriks/bartender/web/functionaltest/ingredient/BddIngredientService.java

```diff
@@ -56,15 +57,21 @@ public Ingredient findIngredient(final String ingredientName) {
         return ingredientRepository.findByName(ingredientName);
     }
 
-    public Ingredient updateIngredient(final String ingredientName, final IngredientType ingredientType){
-        final int ingredientFromRepoId = findIngredient(ingredientName).getId();
-        final Ingredient ingredientToUpdate = new Ingredient(ingredientFromRepoId, ingredientName, ingredientType);
-        ingredientRepository.save(ingredientToUpdate);
-        return ingredientToUpdate;
+    public void updateIngredient(final String originalIngredientName, final String originalIngredientType,
+                                       final String updatedIngredientName, final String updatedIngredientType){
+        List<String> ingredientUpdates = new ArrayList<>();
+        ingredientUpdates.add(originalIngredientName);
+        ingredientUpdates.add(originalIngredientType);
+        ingredientUpdates.add(updatedIngredientName);
+        ingredientUpdates.add(updatedIngredientType);
+        mockMvc.put("/ingredients", ingredientUpdates);
```

Comment:
This is an interesting approach... but as I mentioned elsewhere, it requires the client and server to know the correct order of items in the list. better to create an `Instrument` and pass that object instead? Just like `post` does:

```
    public void addIngredientViaApi(final Ingredient ingredient) {
        mvcResult = mockMvc.postViaApi("/api/ingredients", ingredient);
    }
```

Reply:
If we pass only one ingredient, how do we know which ingredient name to update (suppose we name change Champagne -> Sparkling Wine)? Unless you pass a list of ingredients (oldIngredient, updatedIngredient).

I like the map idea.

Reply:
in REST you update one resource at a time. The URI specifies the resource. For example:

`/people/marie`
`/people/name=marie`
`/books/isbn/9786553932579`
`/books?isbn=9786553932579`

all refer to a specific resource/entity/instance.

once you can target a specific resource, then you can perform CRUD actions on it.

Reply:
So should I structure it like this, if using a map:
```  
public void updateIngredientViaApi(final Ingredient originalIngredient, final Ingredient updatedIngredient){
        Map<Ingredient, Ingredient> ingredientUpdatesMapping = new HashMap<>();
        ingredientUpdates.put(originalIngredient, updatedIngredient);
        mockMvc.put("/api/ingredients", ingredientUpdatesMapping );
    }
```

I'm operating under the assumption an api user does not know the ID